docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,124 @@
+
+Vertical Lagrangian-remap method
+======================
+
+date: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+
+
+Summary
+-------
+
+The vertical Lagrangian-remap method is to be implemented to support the choice 
+of hybrid vertical coordinates. It can be seen as an extension of the Arbitrary 
+Lagrangian-Eulerian method which is already implemented in MPAS-Ocean, a key 
+difference being that the solution of diasurface transport is coordinate-free. 
+This has the advantage of removing the vertical CFL constraint, which is of 
+particular concern for an evolving ice shelf geometry in terrain-following 
+coordinate schemes. Generalization of the vertical coordinate in the horizontal 
+equations of motion is not in the scope of this development. This development 
+will include some infrastructure for specifying the target grid for regridding. 
+The minimum criteria for success are not degrading the accuracy of the solution 
+or the performance by more than [an acceptable threshold] for several basic test 
+cases... .
+
+*The purpose of this section is to summarize what capability is to be added to
+MPAS-Model through this design process. It should be clear what new code will do
+that the current code does not. Summarizing the primary challenges with respect
+to software design and implementation is also appropriate for this section.
+Finally, this statement should contain a general statement with regard to what
+is "success."*
+
+
+Requirements
+------------
+
+Requirement: name-of-requirement-here
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2020/11/24
+
+Contributors: Darren Engwirda, Xylar Asay-Davis, Carolyn Begeman
+
+Requirement: A grid that relaxes to an analytical expression for a target grid.
+
+Requirement: The ability to implement hybrid coordinate systems.
+
+Requirement: XX order accuracy of the solution.
+
+Requirement: Performance is not degraded below an acceptable level.
+
+Requirement: Layer thicknesses are non-negative (and above a minimum thickness).
+
+*Possible additional requirements:
+- re. development of instabilities
+- re. conservation/monotonicity of interpolants
```

Comment:
Ooh, good ones!  Conservation and monotonicity should both be added.  I would think we would require conservation in an integral sense (e.g. column volume is the same to machine precision before and after remapping).  Maybe there would be other requirements (e.g. conservation in a layer-wise sense, or at least the option for it).  @dengwirda, what is possible with PPR?

Regarding monotonicity, @dengwirda would be in a better position than I am to suggest the requirements.  But we would certainly want at least user control over now monotonic the reconstruction is.

Reply:
My understanding of PPR is that there is conservation over the layer integral.

Reply:
I'm not sure how detailed it makes sense to get about the details of the remapping operation in the requirements section. Maybe just include layer-wise conservation?

Reply:
@cbegeman, @xylar, yes, the remapping methods should conserve the vertical integrals of whatever is being remapped (sum(h_k * psi_k) for example, where h_k is layer thickness and psi_k is the layer mean value of a given variable).

There are a few different options for order-of-accuracy + monotonicity / slope-limiting in PPR: PLM (2nd-order), PPM (3rd-order), PQM (5th-order), and either monotone or WENO flux / slope limiting.

I agree that there's no need to get into all of these details in the doc / requirements, but just for reference:

- PLM: Piecewise Linear Method
- PPM: Piecewise Parabolic Method
- PQM: Piecewise Quartic Method
- Monotone limiting: No under/overshoots, but perhaps more numerical diffusion than is desired, can limit order-of-accuracy to 2nd-order.
- WENO: Weighted Essentially Non-oscillatory limiting: "eps"-sized under/overshoots, less numerical diffusion, should generally achieve full order-of-accuracy.